index.js

```diff
@@ -135,6 +134,12 @@ const COMPRESSIONS = {
 class Cache {
   constructor(key, _) {
     const options = _ || {};
+    let tmpdir = path.join(os.tmpdir(), "async-disk-user");
+    try {
+      tmpdir = path.join(os.tmpdir(), username());
+    } catch (e) {
+      // use default tmpdir on username failure
```

Comment:
I have some questions:

0) if `username-sync` has an issue, maybe we should instead fix it there https://github.com/stefanpenner/username-sync
1) I would like to understand the failure mode here, could you explain?
2) in-order to prevent genuine bugs from sneaking in, my preference is for catch clauses to be explicitly targeted at specific errors (expected errors) , re-throwing the exceptional;

```js
catch (e) {
  // let's ensure our catch clauses are specific
   if (e !== null && typeof e === 'object' && e.code === 'ENONT') {
      // comment describing why this is handled
   } else {
     throw e; // re-throw, because something unexpected happened
   }
}
```

Reply:
I do have an open PR for username-sync [here](https://github.com/stefanpenner/username-sync/pull/4)

Reply:
@glharper oo, i didn't notice that. I much prefer that PR. Mind if we close this one, and work to land that one?